### ISSUE TITLE
Add Redirect-to-URL functionality for Pages

### ISSUE THREAD

**Issue by mirie** [forem/forem#18752]
## User Story (Who / What / Why)

As an admin, I’d like to be able to set up a redirect-to-URL so that I can more effectively move people around my Forem.

## Definition of Done

- [ ]  A new option is added to `/admin/customization/pages/new` that allows me to easily set up a URL `301` redirect

## Additional Context

[https://github.com/forem/forem/discussions/16117](https://github.com/forem/forem/discussions/16117)

**Comment by mirie** [comment]
@benhalpern: I have a question about the expectation for how the admin portion should work. I recorded a Loom: https://www.loom.com/share/f29a48f13d8b46cb9db4aa8df671058e

Basically, my question is if the Redirect to URL field is added, say...above the Template field, what is the expectation (I just need to fill out this field and ignore the others?)

**Comment by mirie** [comment]
Notes from Refinement meeting: 
We have hard-coded redirects already setup. Example, https://dev.to/welcome goes to the latest/most recent welcome thread. We want to do this more easily without having to hard-code. Example use case: We want to be able to create new display ads, create a redirect like `/memeMonday` >> redirect to a specific page (example, latest Meme Monday), so that we don't have to update Display Ads every week with the url to the latest meme monday post

Q: Instead of lumping into pages, perhaps we create new functionality?

@pkfrank and @anujbhavsar96: can you sync re: admin side form layout/interactions?

**Comment by anujbhavsar96** [comment]
So just to be sure, @mirie , here is what I understood. Correct me if I’m wrong,

Initially, we wanted to redirect pages to external URLs. But instead, we are building a distinct feature of redirection, right?

If that’s the case - What could be the technical requirements/inputs required from the user/admins? I guess it would be the `Slug` and `Redirection URL`, right? And I see these functions reside at
`/admin/customization/redirection`

**Comment by mirie** [comment]
@anujbhavsar96 That is what was brought up in the refinement discussion -- that maybe we risk overloading too much functionality into this one area, and perhaps we should consider separating out these two different pieces of functionality / areas in admin:
(1) Overriding an existing template / Adding a HTML based template (Pages)
(2) Redirects 

`/admin/customization/redirection` as path (and also separate section) makes sense.

In terms of fields:
- Current path (I'm guessing that is the Slug in what you wrote above)
- Redirection to (Redirection URL)

I think we can skip a dropdown field for various 3xx statuses because we just want 301 Moved Permanently

**Comment by mirie** [comment]
@pkfrank Do you agree with this direction for Redirect implementation?

**Comment by mirie** [comment]
**Refinement Discussion:**

**@benhalpern can you take a look at this issue and see if you are aligned on this approach? Separating it out from Pages vs folding it into Pages.**

- If we fold into Pages, we'll need a design input

Related Issue (cc @lightalloy):
- This issue https://github.com/forem/forem/issues/2914 is related as it touches slugs. Putting on Pages (or something else) feels hard with the way we manage slugs right now (podcasts, articles, pages). Should how we handle redirects be thought of in parallel with how we are thinking about refactoring the StoriesController? With the refactor of the StoriesController, this could change how this is approached --> It feels like the way StoriesController is being refactored, it will probably promote Slugs as some kind of higher order object (being able to find an item via its slug)

Q: StoriesController's job is to handle the logic for Articles. Should it have any insight into things that are Pages, etc? StoriesController shouldn't handle anything that's related to Pages 
A: StoriesController handles a lot, most of the routing work is in one action (based on if/else condition checks on the slug in each of these categories and makes some assumptions):
  - homepage
  -  latest articles
  - individual article
  - user profile
  - all the comments from a user comments (page)
  - organization profile page
  - podcast (single episode and list of episodes)

StoriesController needs to be broken up into several individual controllers that handles different responsibilities, however, because the StoriesController has been the way it has for so long, it IS essentially the essential router for the site. The if/else logic serves as the defacto routing instructions, but these are more implicit rather than explicit. Have to unpack the controller to understand how routing happens in the app.

Even if the StoriesController is split up into individual controllers with more limited responsibilities, there will still need to be a main router controller (that the storiesController is currently acting as) to route generic slug to connect to the proper controller. If we had redirect functionality we would need to  slot this in here, as kind of a switchboard operation. We might want to have 2 kinds of actions here:
(1) one kind of internal render (not an actual 302) where we shell over to the other controller to handle to actual action
and 
(2) a proper 302, where the URL updates, get a 302, etc

- IF we do this work first, then the redirect work would have to live in the StoriesController
- IF we do this work after refactoring the StoriesController, then it will need to live in the a new routing Controller. If we can address the storiesController refactor first, it could make implementing this redirect functionality _potentially_ trivial.

Related conversation re: StoriesController refactor: https://forem-team.slack.com/archives/C7GE84DEJ/p1669727030640249
- @lightalloy thought about having a class to handle which link should belong to a controller -- **@lightalloy can you share with the team more about what you're thinking/where you are on this?**
- @maestromac mentioned the Github has a slug model: Everything in the app has a slug, and the slug determines what will be rendered

Concern: Having a router controller...wouldn't that mean every request would end up being a 302?
- Don't know that the router controller has to redirect everything 302 style
- We probably want our urls to be a generic slug that then renders correctly
- Having this object would allow us to make choices on supporting 302. An example might be podcasts -- the slug is what you're looking at, maybe it would be valuable to have that 302 to a new permanent address for podcasts that's more explicit, and maybe this doesn't have to go through the routing controller. Right now we can't do this because there is no routing controller to go through.
- It sounds like this is what we want to do for some of these static pages: an Internal render and not an actual 302
- We'll probably find slugs are not as useful as we want (this has happened)
- **What is the SEO implication on this method?**
